json/src/value.rs

```diff
@@ -426,13 +446,23 @@ impl de::Deserialize for Value {
                 Ok(Value::Array(values))
             }
 
+            #[cfg(not(feature = "preserve_order"))]
             #[inline]
             fn visit_map<V>(&mut self, visitor: V) -> Result<Value, V::Error>
                 where V: de::MapVisitor,
             {
                 let values = try!(de::impls::BTreeMapVisitor::new().visit_map(visitor));
                 Ok(Value::Object(values))
             }
```

Comment:
I think this would be clearer as a single method containing:

```rust
#[cfg(not(feature = "preserve_order"))]
type Visitor<K, T> = de::impls::BTreeMapVisitor<K, T>;
#[cfg(feature = "preserve_order")]
type Visitor<K, T> = linked_hash_map::serde::LinkedHashMapVisitor<K, T>;

let values = try!(Visitor::new().visit_map(visitor));
Ok(Value::Object(values))
```

That makes it clear they are doing exactly the same thing just with different visitor types.